src/preconditions/Cooldown.ts

```diff
@@ -0,0 +1,57 @@
+import { Precondition, PreconditionContext } from '../lib/structures/Precondition';
+import { BucketType } from '../lib/types/Enums';
+import type { Message } from 'discord.js';
+import type { Command } from '../lib/structures/Command';
+import { Bucket } from '@sapphire/ratelimits';
+
+interface CooldownOptions {
+	delay: number;
+	limit: number;
+	bucketType: BucketType;
+}
+
+export class CorePrecondition extends Precondition {
```

Comment:
```suggestion
export class CooldownPrecondition extends Precondition {
```
since its not a default export

Reply:
For consistency, we call all core preconditions CorePrecondition.

If you eval (using Klasa's eval), the last thing you want is a list of all preconditions as the type, `PreconditionStore<CorePrecondition | UserPrecondition>` looks a lot nicer.

Reply:
Yes but won't that create issues when importing? or are they exported with a proper name when re-exported from the index

Reply:
Why would it?